### PR TITLE
fix macro when ignore_index is set

### LINKED ISSUE #1691: Accuracy is wrong when ignore_index is set
## 🐛 Bug

When the _ignore\_index_ argument is set to a value, the MulticlassAccuracy object sets the class accuracy to zero, but considers it in the average calculation.

### To Reproduce

```python
import torch
from torchmetrics.classification import MulticlassAccuracy

# simulate the output of a perfect predictor (i.e. preds == target)
target = torch.tensor([0, 1, 2, 0, 1, 2])
preds  = target

metric = MulticlassAccuracy(num_classes=3, average='none', ignore_index=0)
res = metric(preds, target)
print(res)
# it prints [0., 1., 1.]

metric = MulticlassAccuracy(num_classes=3, average='macro', ignore_index=0)
res = metric(preds, target)
print(res)
# it prints 0.6667 instead of 1
```

### Expected behavior

It should not take into account the ignored class.

### Environment

- TorchMetrics version: 0.11.4 installed from pip
- Python & PyTorch Version: 3.10.6 - 1.13.0
- Any other relevant information such as OS (e.g., Linux): Ubuntu 22.04

### PR DESCRIPTION
**Pull request by edumotya**
## What does this PR do?

Fixes #1691

Additionally, tests have been updated to exclude the ignored class from the macro accuracy averaging.

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/torchmetrics/functional/classification/stat_scores.py [hunk 1]
```diff
@@ -416,6 +416,8 @@ def _multiclass_stat_scores_update(
         fp = confmat.sum(0) - tp
         fn = confmat.sum(1) - tp
         tn = confmat.sum() - (fp + fn + tp)
+        if ignore_index is not None:
+            fp[ignore_index] = 0
```
Comment (SkafteNicki) [inline src/torchmetrics/functional/classification/stat_scores.py:420]: this is probably correct, but since many metrics derive from the `stat_scores` class that means that basically all would need to have their unittests fixed

**Comment by Borda** [comment]
@edumotya seems like most of the tests are off; could you pls check it...
turning it back to draft until most other checks are green... pls make it as ready to review when you resolve them... :flamingo:

**Comment by edumotya** [comment]
sure @Borda, I am sorry for the late response. It might take me some time, it is not straightforward to fix the affected tests, I'll do my best

**Comment by ding3820** [comment]
Any update?

**Comment by Borda** [comment]
> Any update?

I think this is related #2710

File: src/torchmetrics/functional/classification/stat_scores.py (hunk 1 above)
Comment (Borda) [reply src/torchmetrics/functional/classification/stat_scores.py:420]: what do you mean by it needs to be fixed?

**Comment by Borda** [comment]
> Any update?

@ding3820 do you want to take it over and use @edumotya's commits in your/new PR?
